internal/gusserver/gusserver.go

```diff
@@ -133,15 +135,17 @@ func newServer(databaseDir string) (*server, *http.ServeMux, error) {
 
 func Main() error {
 	var (
-		listen      = flag.String("listen", "localhost:8655", "[host]:port listen address")
-		databaseDir = flag.String("database_dir", "/var/lib/gus", "database directory for GUS internal state. the special value :memory: stores state in memory")
+		listen         = flag.String("listen", "localhost:8655", "[host]:port listen address")
+		databaseType   = flag.String("database_type", "sqlite", "can be one of: sqlite, postgres")
+		databaseSource = flag.String("database_source", ":memory:", "database source for GUS internal state. can be :memory: (default. stores state in memory), directory path (sqlite) or an connection DSN (postgres. reference: https://pkg.go.dev/github.com/lib/pq#hdr-Connection_String_Parameters)")
 	)
 	flag.Parse()
 
-	if *databaseDir != ":memory:" {
-		*databaseDir = filepath.Join(*databaseDir, "gus.db")
+	if *databaseType == "sqlite" {
```

Comment:
I think you need both conditionals here: *databaseType == "sqlite" && *databaseDir != ":memory:"